### PR TITLE
Refactor/replace some ff with flags

### PR DESCRIPTION
**Pull request by jskelin**
#### What this PR does / Why we need it:
Direct usage of feature flags in `pkg` prevents `pkg` to be reusable as a library. This PR replaces some usages of `MONACO_FEAT_DOWNLOAD_FILTER` and  `MONACO_FEAT_DOWNLOAD_FILTER_CLASSIC_CONFIGS` with the flags

### CODE REVIEW

File: pkg/download/classic/downloader.go [hunk 1]
```diff
@@ -42,9 +41,11 @@ type (
 	Downloader struct {
 		apisToDownload api.APIs
 
+		filter bool
```
Comment (UnseenWizzard) [inline pkg/download/classic/downloader.go:44]: please ensure the `NewDownloader` constructor sets this to `true` by default, before applying any options. We also assume the feature flags are ON by default in this case.

File: cmd/monaco/download/downloaders.go [hunk 2]
```diff
@@ -89,6 +90,10 @@ func removeSkipDownload(api api.API) bool {
 	return false
 }
 
+func ShouldApplyFilter() bool {
```
Comment (UnseenWizzard) [inline cmd/monaco/download/downloaders.go:93]: I don't think this should be exported anymore now

File: pkg/download/classic/downloader_test.go [hunk 3]
```diff
@@ -14,14 +14,16 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package classic
+package classic_test
 
 import (
 	"context"
 	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/cmd/monaco/download"
```
Comment (UnseenWizzard) [inline pkg/download/classic/downloader_test.go:22]: even for tests, I don't think a package should ever import from cmd, please remove that (think it's to access the feature flag convenience method, IMO that one's not needed for testing. Currently we assume they are ON so keeping the tests at default filtering or where needed just explicitly setting WithFiltering(true) or WithFiltering(false) should be enough. 

Testing that the flags control the option should be done on the cmd package integration test IMO.

**Review by UnseenWizzard — CHANGES_REQUESTED** [review]

File: cmd/monaco/download/downloaders.go (hunk 2 above)
Comment (jskelin) [reply cmd/monaco/download/downloaders.go:93]: I agree. To avoid massive changes in tests, this is exported. Is it enough to write a comment?

Comment (UnseenWizzard) [reply cmd/monaco/download/downloaders.go:93]: Do the tests ever set the environment variables though?
From what I saw, only `TestDownload_SkipConfigBeforeDownload` does actually set the env vars, all other uses could set the value they need explicitly or just default to true. 

For `TestDownload_SkipConfigBeforeDownload` IMO on the downloader level this should change to just verify the boolean option works as intended - defaulting to true and possible to turn off. 

That the env vars translate to the correct input for the downloader that is created in the cmd package should IMO be tested there, not by tying the logic of cmd and pkg together. 
I think the test could move to the cmd/ tests easily as it just seems to use public methods of the downloader anyway?

File: pkg/download/classic/downloader.go (hunk 1 above)
Comment (warber) [reply pkg/download/classic/downloader.go:44]: if that's the reasonable default, you could turn it around, and change it to `skipFilter bool`

File: pkg/download/classic/downloader.go [hunk 4]
```diff
@@ -74,12 +76,18 @@ func WithAPIs(apis api.APIs) ConstOption {
 	}
 }
 
-func WithAPIContentFilters(apiFilters map[string]contentFilter) ConstOption {
+func WithAPIContentFilters(apiFilters map[string]ContentFilter) ConstOption {
```
Comment (warber) [inline pkg/download/classic/downloader.go:79]: just wondering, what does the "Const" mean in "ConstOption" i don't get it :D

File: pkg/download/classic/downloader.go [hunk 5]
```diff
@@ -74,12 +76,18 @@ func WithAPIs(apis api.APIs) ConstOption {
...
 	return func(d *Downloader) {
 		d.apiContentFilters = apiFilters
 	}
 }
 
+func WithFiltering(b bool) ConstOption {
```
Comment (warber) [inline pkg/download/classic/downloader.go:85]: just wondering, what does the "Const" in "ConstOption" mean? I don't get it :D

Comment (jskelin) [reply pkg/download/classic/downloader.go:85]: from `Construction`
The name is far from perfect. Just propose a better one...

Comment (warber) [reply pkg/download/classic/downloader.go:85]: `Option`, if that does not work, `DownloaderOption` or no name at all but just the signature

Comment (jskelin) [reply pkg/download/classic/downloader.go:85]: Typing is forced to have a compile-time error if the signature of `func NewDownloader(client dtclient.Client, opts ...ConstOption) *Downloader` is changed!

I'll rename it to `Option`

**Review by UnseenWizzard — APPROVED** [review]

**Pull request merged**